src/servers/trigger.ts

```diff
@@ -115,26 +121,53 @@ export async function onchainTransactionEventHandler(tx) {
       throw error
     }
 
-    if (tx.is_confirmed === false) {
+    // we only handle pending notification here because we wait more than 1 block
+    if (!tx.is_confirmed) {
       onchainLogger.info(
         { transactionType: "receipt", pending: true },
         "mempool appearence",
       )
-    } else {
-      // onchain is currently only BTC
-      const wallet = await WalletFactory({ user, logger: onchainLogger })
-      await wallet.updateOnchainReceipt()
+
+      await transactionNotification({
+        type: "onchain_receipt_pending",
+        user,
+        logger: onchainLogger,
+        amount: Number(tx.tokens),
+        txid: tx.id,
+      })
     }
+  }
+}
 
-    const type = tx.is_confirmed ? "onchain_receipt" : "onchain_receipt_pending"
-    await transactionNotification({
-      type,
-      user,
-      logger: onchainLogger,
-      amount: Number(tx.tokens),
-      txid: tx.id,
-    })
+export async function onchainBlockEventhandler({ lnd, height }) {
+  const lookBack = ONCHAIN_MIN_CONFIRMATIONS + 1
+  const onchainTxns = await getOnChainTransactions({ lnd, lookBack, incoming: true })
+  const hasTransactions = onchainTxns && onchainTxns.length
+
+  if (!hasTransactions) {
+    logger.info(`finish block ${height} handler without transactions`)
```

Comment:
maybe the wording could be improve here? `no transaction to handle, skipping block ${height}`?

having "finish" implied to me we did some work but we didn't really

Reply:
Changed https://github.com/GaloyMoney/galoy/pull/428/commits/20668101e2a16ba4ad949eaa1af8980a7bbc26ed